File: ios/plugin/ios/Plugin/Sources/MorphProtocol/MorphClient.swift
import Foundation

/// Main MorphProtocol client facade
/// Provides a simple API for starting and stopping the client
public class MorphClient {
    private let udpClient: MorphUdpClient
    
    public var onConnected: (() -> Void)? {
        get { udpClient.onConnected }
        set { udpClient.onConnected = newValue }
    }
    
    public var onDisconnected: (() -> Void)? {
        get { udpClient.onDisconnected }
        set { udpClient.onDisconnected = newValue }
    }
    
    public var onError: ((Error) -> Void)? {
        get { udpClient.onError }
        set { udpClient.onError = newValue }
    }
    
    public init(config: ClientConfig) {
        self.udpClient = MorphUdpClient(config: config)
    }
    
    /// Start the client
    public func start() {
        udpClient.start()
    }
    
    /// Stop the client
    public func stop() {
        udpClient.stop()
    }
}

File: android/plugin/android/src/main/java/com/morphprotocol/client/MorphClient.kt
package com.morphprotocol.client

import android.content.Context
import com.morphprotocol.client.config.ClientConfig
import com.morphprotocol.client.network.MorphUdpClient

/**
 * Connection result data class.
 */
data class ConnectionResult(
    val success: Boolean,
    val serverPort: Int = 0,
    val clientId: String = "",
    val message: String = ""
)

/**
 * Main MorphProtocol client facade.
 * Provides a simple API for starting and stopping the client.
 */
class MorphClient(
    private val config: ClientConfig,
    private val context: Context
) {
    private val udpClient = MorphUdpClient(config, context)
    
    /**
     * Start the client (blocking).
     * Should be called from a background thread in the service.
     * 
     * @param onConnected Callback invoked when connection is established
     */
    fun start(onConnected: ((ConnectionResult) -> Unit)? = null): ConnectionResult {
        return udpClient.start(onConnected)
    }
    
    /**
     * Stop the client.
     */
    fun stop() {
        udpClient.stop()
    }
}


